src/include/buffer.h

```diff
@@ -865,7 +865,9 @@ struct error_code;
 	if (first_round) {
 	  impl_f(first_round);
 	}
-	if (const auto second_round = len - first_round; second_round) {
+	// no C++17 for the sake of the C++11 guarantees of librados, sorry.
+	const auto second_round = len - first_round;
```

Comment:
👍 for the fix. but the comment itself will only be confusing, because it's referring to code that _used_ to exist

Reply:
> +1 for the fix. but the comment itself will only be confusing, because it's referring to code that _used_ to exist

@cbodley @rzarzynski I added the "DNM" tag since this has already gone through QA, just so it doesn't accidentally get merged before this comment is addressed. Please remove the DNM tag as necessary.

Reply:
i don't want to delay merge if it's ready. i'll gladly follow up with a separate PR if that's easier